tbdex/rfq/rfq.go

```diff
@@ -200,4 +193,37 @@ func (r RFQ) Digest() ([]byte, error) {
 	return hashed, nil
 }
 
+func (r *RFQ) verifyPrivateData(strict bool) error {
+	if r.PrivateData == nil {
+		if strict {
+			return errors.New("private data is missing")
+		}
+
+		return nil // no PrivateData to verify
+	}
+
+	if strict || r.Data.ClaimsHash != "" {
+		payload := []any{r.PrivateData.Salt, r.PrivateData.Claims}
+		if err := tbdex.VerifyDigest(r.Data.ClaimsHash, payload); err != nil {
+			return fmt.Errorf("failed to verify claims: %w", err)
+		}
+	}
+
+	if strict || r.Data.Payin.PaymentDetailsHash != "" {
```

Comment:
if `PaymentDetailsHash` is null, then we shouldn't validate. that means that the payin has no payment details, which is still valid for RFQs.